sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -73,46 +127,31 @@ case class OrderedDistribution(ordering: Seq[SortOrder]) extends Distribution {
       "An AllTuples should be used to represent a distribution that only has " +
       "a single partition.")
 
-  // TODO: This is not really valid...
-  def clustering: Set[Expression] = ordering.map(_.child).toSet
+  override def requiredNumPartitions: Option[Int] = None
```

Comment:
Out of curiosity, should an `OrderedDistribution` make any guarantees around clustering? Do we care if "tuples that share the same value for the ordering expressions will never be split across partitions"?

Reply:
According to the comment `This is a strictly stronger guarantee than [[ClusteredDistribution]]`, we want to guarantee it. However we actually don't respect it, see https://github.com/apache/spark/pull/19080/files#r136419947

Since it is an existing problem, I'd like to fix it in another PR.